tensorflow_addons/metrics/matthews_correlation_coefficient.py

```diff
@@ -0,0 +1,151 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Matthews Correlation Coefficient Implementation."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+from tensorflow.keras.metrics import Metric
+import numpy as np
+
+
+class MatthewsCorrelationCoefficient(Metric):
+    """Computes the Matthews Correlation Coefficient.
+
+    The statistic is also known as the phi coefficient.
+    The Matthews correlation coefficient (MCC) is used in
+    machine learning as a measure of the quality of binary
+    and multiclass classifications. It takes into account
+    true and false positives and negatives and is generally
+    regarded as a balanced measure which can be used even
+    if the classes are of very different sizes. The correlation
+    coefficient value of MCC is between -1 and +1. A
+    coefficient of +1 represents a perfect prediction,
+    0 an average random prediction and -1 an inverse
+    prediction. The statistic is also known as
+    the phi coefficient.
+
+    MCC = (TP * TN) - (FP * FN) /
+          ((TP + FP) * (TP + FN) * (TN + FP ) * (TN + FN))^(1/2)
+
+    Args:
+       num_classes : Number of unique classes in the dataset.
+
+    Returns:
+       mcc : float (the Matthews correlation coefficient)
+
+    Usage:
+    ```python
+    actuals = tf.constant([[1.0], [1.0], [1.0], [0.0]],
+             dtype=tf.float32)
+    preds = tf.constant([[1.0], [0.0], [1.0], [1.0]],
+             dtype=tf.float32)
+    # Matthews correlation coefficient
+    mcc = MatthewsCorrelationCoefficient(num_classes=1)
+    output.update_state(actuals, preds)
+    print('Matthews correlation coefficient is:',
+    output.result().numpy())
```

Comment:
I guess this should be `mcc` instead of `output`.

Reply:
Yeah, it is. Thanks @WindQAQ.